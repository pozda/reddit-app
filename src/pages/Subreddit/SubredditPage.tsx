import { AxiosResponse } from "axios";
import { useState, useEffect } from "react";
import { useParams } from "react-router-dom";
import {RESTService as NetworkService} from "../../network/NetworkService"
import { getLatestThreads } from "../../network/Requests";



const networkService = new NetworkService();

const SubredditPage:React.FC<{}> = () => {
  const [latestThreads, setLatestThreads] = useState()
  const [selectedSub, setSelectedSub] = useState()
  // @ts-ignore
  const {subredditId} = useParams()
  const fetchLatestThreads = (id: string) => {
    const config = getLatestThreads(id)
    networkService.makeRequest(config)
    .then((response: AxiosResponse<any>) => { 
        setLatestThreads(response.data)
        setSelectedSub(subredditId)
    }) 
  }


useEffect(()=>{
  fetchLatestThreads(subredditId!)
  // eslint-disable-next-line
},[subredditId])


    return (
        <>
        <h1>selected sub: {selectedSub}</h1>
        {// @ts-ignore
latestThreads && latestThreads?.children?.map(subs => {
return (
<a href={subs.data.url} key={subs.data.title}><h3>{subs.data.title}</h3></a>
);
})
}
      </>
    )
}

export default SubredditPage