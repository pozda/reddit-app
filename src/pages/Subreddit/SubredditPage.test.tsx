import React from 'react'
import {shallow} from 'enzyme'
import {Router} from 'react-router-dom'
import {createMemoryHistory} from 'history'
import SubredditPage from './SubredditPage'

describe('SubredditPage', () => {
    it('should be defined', () => {
        expect(SubredditPage).toBeDefined()
    })
    it('should render correctly', () => {
        const history = createMemoryHistory()
        const route = '/r/jesttest'
        history.push(route)
        const wrapper = shallow(
            <Router history={history}>
                <SubredditPage />
            </Router>
        )
        expect(wrapper).toMatchSnapshot()
    })
})