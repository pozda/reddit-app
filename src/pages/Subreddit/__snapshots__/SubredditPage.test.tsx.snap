// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubredditPage should render correctly 1`] = `ShallowWrapper {}`;
